stdlib/2and3/tarfile.pyi

```diff
@@ -30,7 +36,7 @@ if sys.version_info < (3,):
     TAR_PLAIN = ...  # type: int
     TAR_GZIPPED = ...  # type: int
 
-def open(name: Optional[str] = ..., mode: str = ...,
+def open(name: Optional[_Path] = ..., mode: str = ...,
```

Comment:
I don't think AnyStr makes sense here, because this is the only argument to the function (AnyStr is a typevar, and it should be used to constrain multiple arguments or an argument and the return value to be the same typE). I think this function can just take `Union[bytes, str]` as an argument; same for most others below.